### ISSUE TITLE
Copying a container inserts deleted elements into the copy of the container

### ISSUE THREAD

**Issue by user123801** [b13/container#244]
Environment:
TYPO3 10.4.26
PHP 7.4
Container 1.2.0 and 1.5.0 (1.2, then upgraded to 1.5, problem still exists)

Error: 
When copying a container from one page to another page or to the same page, it inserts deleted content elements from an unrelated page into the container (or from the same page if the target page happens to be the unrelated page). The deleted content elements always seem to be from one specific page regardless of the source/target page of the container. The deleted content elements are textmedia elements with and without file references.

We tried deleting the content elements entirely in the recycler, this resolved it for a while before it started inserting other deleted content elements.

**Comment by achimfritz** [comment]
Hi @user123801 ,

i have try to reproduce the error, having
```
mysql> select CType,uid,tx_container_parent,deleted,hidden from tt_content;
+---------------------------------+-----+---------------------+---------+--------+
| CType                           | uid | tx_container_parent | deleted | hidden |
+---------------------------------+-----+---------------------+---------+--------+
| b13-2cols-with-header-container |   1 |                   0 |       0 |      0 |
| textmedia                       |   2 |                   1 |       1 |      0 |
+---------------------------------+-----+---------------------+---------+--------+
```
and copied uid 1 to same page, results in
```
mysql> select CType,uid,tx_container_parent,deleted,hidden from tt_content;
+---------------------------------+-----+---------------------+---------+--------+
| CType                           | uid | tx_container_parent | deleted | hidden |
+---------------------------------+-----+---------------------+---------+--------+
| b13-2cols-with-header-container |   1 |                   0 |       0 |      0 |
| textmedia                       |   2 |                   1 |       1 |      0 |
| b13-2cols-with-header-container |   3 |                   0 |       0 |      1 |
+---------------------------------+-----+---------------------+---------+--------+
```

Can you give more information how to reproduce?

**Comment by user123801** [comment]
Hi @achimfritz,

thanks for your fast reply.

Source container (the content elements in the container)

mysql> select uid, pid, colPos, hidden, deleted from tt_content where tx_container_parent=14291;
+-------+-----+--------+--------+---------+
| uid   | pid | colPos | hidden | deleted |
+-------+-----+--------+--------+---------+
| 14292 | 971 |    201 |      1 |       0 |
| 14293 | 971 |    201 |      1 |       0 |
| 14294 | 971 |    201 |      1 |       0 |
| 14295 | 971 |    202 |      1 |       0 |
| 14296 | 971 |    202 |      1 |       0 |
| 14297 | 971 |    202 |      1 |       0 |
| 14298 | 989 |    201 |      0 |       0 |
| 14299 | 989 |    202 |      0 |       0 |
+-------+-----+--------+--------+---------+

This is what the copied container (the content elements in the container) looks like:

mysql> select uid, pid, colPos, hidden, deleted from tt_content where tx_container_parent=15696;
+-------+------+--------+--------+---------+
| uid   | pid  | colPos | hidden | deleted |
+-------+------+--------+--------+---------+
| 15697 | 1100 |    201 |      1 |       0 |
| 15698 | 1100 |    201 |      1 |       0 |
| 15699 | 1100 |    201 |      1 |       0 |
| 15700 | 1100 |    201 |      1 |       0 |
| 15701 | 1100 |    202 |      1 |       0 |
| 15702 | 1100 |    202 |      1 |       0 |
| 15703 | 1100 |    202 |      1 |       0 |
| 15704 | 1100 |    202 |      1 |       0 |
+-------+------+--------+--------+---------+

So it seems I was wrong about the actual problem. There are a number of hidden elements in the container I want to copy. Those are not visible in the container in the backend for some reason. Those are visible in the container after copying it.

Most likely, our editors didn't delete the content elements they no longer needed. This is then most likely not an issue with the extension. The actual problem is that the hidden content elements in the source container are not shown in the TYPO3 backend, then all are visible after copying/inserting the container to any other page.

I think this issue can be closed. Thanks for your time.

**Comment by achimfritz** [comment]
Hi @user123801 ,

your problem looks like a bug i also have detected some time ago. On EXT:container Version <=1.3 (or 1.2, not sure), there was a bug, when a container is copied to another page at top. Results, the Child-Elements has wrong pid, but they are not shown in BE as unused elements (they are "invisible" in BE).

I have already a branch, that fix this issue (this elements appears on the wrong page as unused elements). There is also an Command in this branch, which deletes all this elements (this command should be harmless to use, because it deletes only CEs that are not visible in BE/FE).

It would be very helpful if you can test the branch and report if this this working for you.

https://github.com/b13/container/commits/task/integrity2 (the branch refactor also some integrity information, you can ignore)

**Comment by user123801** [comment]
@achimfritz We already fixed it in the database this time, but sure, I'll keep an eye out for this. If it happens again, I'll be sure to test the branch.

**Comment by achimfritz** [comment]
fixed in 1.6.0 (with https://github.com/b13/container/commit/053752756d1981d1f5034c17e95448997eb55303)
use "container:integrity" command to find unconsisent CE
use "container:deleteChildrenWithWrongPid" command to the this
